src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java

```diff
@@ -127,10 +131,11 @@ public UserRole getUserRoleClassifications(String userRole) {
     }
 
     @Override
+    @Cacheable("userRolesCache")
```

Comment:
This method makes a call to the definition store to retrieve security classification associated with the roles. I think the security classification mapping on a role does not change often and this can be cached.

Reply:
How long is this cached?  I assume a change on def-store i.e. new definition would be the only way for this to change.  Do we have anything from def-store that trigger a cache invalidation on data-store?

Reply:
Its cached for 72 hours by default. Currently there is no mechanism for invalidating cache. What the endpoint does is retrieve security classifications for the roles passed. I assume the security classification on roles don't change. When we add a role, we assign security classification. I am not aware if security classification on a role can be changed. @mario-paniccia may know.

Reply:
I think we can refactor this to make sure caching works to its full potential. Let me explain. Imagine we have 10 user roles, each with its own classification. Because we try to cache the result of requesting 
a list of user roles, this means having to cache a huge number of potential combinations of those 10 roles. This mean lower probability the cached value is used in the future (cache hit).
Instead we could refactor the method to accept one single user role. Once we do that, at most the cache will cache 10 entries. This will increase the cache hit percentage.

Reply:
good point, done.